here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/DefaultViewHandler.java

```diff
@@ -0,0 +1,138 @@
+
+import static com.here.naksha.lib.handlers.AbstractEventHandler.EventProcessingStrategy.*;
+
+import com.here.naksha.lib.core.IEvent;
+import com.here.naksha.lib.core.INaksha;
+import com.here.naksha.lib.core.NakshaContext;
+import com.here.naksha.lib.core.models.XyzError;
+import com.here.naksha.lib.core.models.naksha.EventHandler;
+import com.here.naksha.lib.core.models.naksha.EventTarget;
+import com.here.naksha.lib.core.models.storage.*;
+import com.here.naksha.lib.core.storage.IReadSession;
+import com.here.naksha.lib.core.storage.IStorage;
+import com.here.naksha.lib.core.storage.IWriteSession;
+import com.here.naksha.lib.core.util.json.JsonSerializable;
+import com.here.naksha.lib.view.IView;
+import com.here.naksha.lib.view.ViewLayer;
+import com.here.naksha.lib.view.ViewLayerCollection;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultViewHandler extends AbstractEventHandler {
+
+  private static final Logger logger = LoggerFactory.getLogger(DefaultViewHandler.class);
+
+  private final @NotNull EventHandler eventHandler;
+  private final @NotNull EventTarget<?> eventTarget;
+  private final @NotNull DefaultViewHandlerProperties properties;
+
+  public DefaultViewHandler(
+      final @NotNull EventHandler eventHandler,
+      final @NotNull INaksha hub,
+      final @NotNull EventTarget<?> eventTarget) {
+    super(hub);
+    this.eventHandler = eventHandler;
+    this.eventTarget = eventTarget;
+    this.properties = JsonSerializable.convert(eventHandler.getProperties(), DefaultViewHandlerProperties.class);
+  }
+
+  @Override
+  protected EventProcessingStrategy processingStrategyFor(IEvent event) {
+    final Request<?> request = event.getRequest();
+    if (request instanceof ReadFeatures
+        || request instanceof WriteXyzFeatures
+        || request instanceof WriteXyzCollections) {
+      return PROCESS;
+    }
+    return NOT_IMPLEMENTED;
+  }
+
+  @Override
+  public @NotNull Result process(@NotNull IEvent event) {
+
+    final NakshaContext ctx = NakshaContext.currentContext();
+    final Request<?> request = event.getRequest();
+    logger.info("Handler received request {}", request.getClass().getSimpleName());
+
+    final String storageId = properties.getStorageId();
+
+    if (storageId == null) {
+      logger.error("No storageId configured");
+      return new ErrorResult(XyzError.NOT_FOUND, "No storageId configured for handler.");
+    }
+    logger.info("Against Storage id={}", storageId);
+    addStorageIdToStreamInfo(storageId, ctx);
+
+    final IStorage storageImpl = nakshaHub().getStorageById(storageId);
+    logger.info("Using storage implementation [{}]", storageImpl.getClass().getName());
+
+    if (storageImpl instanceof IView view) {
+
+      view.setViewLayerCollection(
+          prepareViewLayerCollection(nakshaHub().getSpaceStorage(), properties.getSpaceIds()));
```

Comment:
With introduction of DefaultViewHandlerProperties.java the things got simplified a bit. However, we still need to check that getSpaceIds is not null and not empty.